pkg/apis/azure/validation/shoot.go

```diff
@@ -31,6 +31,12 @@ import (
 
 const maxDataVolumeCount = 64
 
+var k8sV121 *semver.Version
+
+func init() {
+	k8sV121 = semver.MustParse("v1.21.0")
```

Comment:
Minor nit: I think it would be better if you would do this with `semver.NewConstraint("<= v1.21")`.